accounts/users.go

```diff
@@ -20,54 +21,70 @@ func CheckPasswordHash(hash, password []byte) bool {
 	return err == nil
 }
 
+// ValidatePassword ensures a password meets minimum security requirements
+func ValidatePassword(password string) bool {
+	var containsNum, containsUpper, containsSpecial bool
+	letters := 0
+
+	for _, c := range password {
+		switch {
+		case unicode.IsNumber(c):
+			containsNum = true
+		case unicode.IsUpper(c):
+			containsUpper = true
+			letters++
+		case unicode.IsPunct(c) || unicode.IsSymbol(c):
+			containsSpecial = true
+		case unicode.IsLetter(c) || c == ' ':
+			letters++
+		default:
+			return false
+		}
+	}
+	validLen := letters >= 6 && letters <= 10
+	return containsNum && containsUpper && containsSpecial && validLen
```

Comment:
Nice. This is really clean. 👌

Reply:
Thanks :)  I tried with regex first, but Golang regex does not allow forward lookups (b/c of performance) so this was my second attempt at it.